compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/reference/PsiToAnnotationSpecHelper.kt

```diff
@@ -0,0 +1,135 @@
+package com.squareup.anvil.compiler.internal.reference
+
+import com.squareup.anvil.compiler.api.AnvilCompilationException
+import com.squareup.anvil.compiler.internal.asClassName
+import com.squareup.anvil.compiler.internal.asClassNameOrNull
+import com.squareup.anvil.compiler.internal.properties
+import com.squareup.anvil.compiler.internal.requireFqName
+import com.squareup.kotlinpoet.CodeBlock
+import com.squareup.kotlinpoet.MemberName
+import org.jetbrains.kotlin.com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.descriptors.ModuleDescriptor
+import org.jetbrains.kotlin.fir.lightTree.converter.nameAsSafeName
+import org.jetbrains.kotlin.lexer.KtTokens
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.psi.KtClass
+import org.jetbrains.kotlin.psi.KtClassLiteralExpression
+import org.jetbrains.kotlin.psi.KtCollectionLiteralExpression
+import org.jetbrains.kotlin.psi.KtElement
+import org.jetbrains.kotlin.psi.KtExpression
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.psi.KtNameReferenceExpression
+import org.jetbrains.kotlin.psi.KtNamedFunction
+import org.jetbrains.kotlin.psi.KtObjectDeclaration
+import org.jetbrains.kotlin.psi.KtProperty
+import org.jetbrains.kotlin.psi.KtQualifiedExpression
+import org.jetbrains.kotlin.psi.psiUtil.containingClass
+
+internal fun KtExpression.codeBlock(module: ModuleDescriptor): CodeBlock {
```

Comment:
It made sense to me to move all these functions that are used for a single operation for `psi.toAnnotationSpec()` into a separate file to isolated them better.